src/mempoollimit.h

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2019 The Zcash developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or https://www.opensource.org/licenses/mit-license.php .
+
+#ifndef MEMPOOLLIMIT_H
+#define MEMPOOLLIMIT_H
+
+#include <map>
+#include <set>
+#include <vector>
+
+#include "boost/optional.hpp"
+#include "primitives/transaction.h"
+#include "uint256.h"
+
+const size_t DEFAULT_MEMPOOL_TOTAL_WEIGHT_LIMIT = 80000000;
+const int64_t DEFAULT_MEMPOOL_EVICTION_MEMORY_MINUTES = 60;
+
+const size_t RECENTLY_EVICTED_SIZE = 40000;
+const uint64_t MIN_TX_WEIGHT = 4000;
+const uint64_t LOW_FEE_PENALTY = 16000; 
+
+class RecentlyEvictedList
+{
+    const size_t capacity;
+    size_t txIdsAndTimesIndex = 0;
+
+    const int64_t timeToKeep;
+    // Pairs of txid and time (seconds since epoch)
+    boost::optional<std::pair<uint256, int64_t>> txIdsAndTimes[RECENTLY_EVICTED_SIZE];
+    std::set<uint256> txIdSet;
```

Comment:
Couldn't both of these data structures be replaced by a `std::deque`? And the code made quite a bit simpler and more efficient? A deque is perfect for implementing a FIFO.

Reply:
I made the change to use a `std::deque` and it did simplify the code and should reduce the memory footprint in normal circumstances.

Using an `unordered_set` rather than a `set` does not compile and I think would require making `uint256` hashable.